server/channels/app/web_broadcast_hooks.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	pUtils "github.com/mattermost/mattermost/server/public/utils"
+	"github.com/mattermost/mattermost/server/v8/channels/app/platform"
+)
+
+const (
+	broadcastAddMentions  = "add_mentions"
+	broadcastAddFollowers = "addfollowers"
+)
+
+func (s *Server) makeBroadcastHooks() map[string]platform.BroadcastHook {
+	return map[string]platform.BroadcastHook{
+		broadcastAddMentions:  &addMentionsBroadcastHook{},
+		broadcastAddFollowers: &addFollowersBroadcastHook{},
+	}
+}
+
+type addMentionsBroadcastHook struct{}
+
+func (h *addMentionsBroadcastHook) ShouldProcess(msg *model.WebSocketEvent, webConn *platform.WebConn, args map[string]any) bool {
+	if msg.EventType() != model.WebsocketEventPosted {
+		return false
+	}
+
+	mentions, ok := args["mentions"].(model.StringArray)
+	if !ok {
+		mlog.Warn("Invalid mentions value passed to addMentionsBroadcastHook", mlog.Any("mentions", args["mentions"]))
+		return false
+	}
+
+	// This hook will only modify the event if the current user is mentioned by the post
+	return pUtils.Contains[string](mentions, webConn.UserId)
+}
+
+func (h *addMentionsBroadcastHook) Process(msg *model.WebSocketEvent, webConn *platform.WebConn, args map[string]any) *model.WebSocketEvent {
+	mentions, ok := args["mentions"].(model.StringArray)
+	if !ok {
+		mlog.Warn("Invalid mentions value passed to addMentionsBroadcastHook", mlog.Any("mentions", args["mentions"]))
+		return msg
+	}
+
+	hasMention := false
+	if len(mentions) > 0 {
+		hasMention = pUtils.Contains(mentions, webConn.UserId)
+	}
+
+	if hasMention {
+		// Note that the client expects this field to be stringified
+		msg.AddWithCopy("mentions", model.ArrayToJSON([]string{webConn.UserId}))
+	}
+
+	return msg
+}
+
+func UseAddMentionsHook(message *model.WebSocketEvent, mentionedUsers model.StringArray) {
+	message.GetBroadcast().AddHook(broadcastAddMentions, map[string]any{
```

Comment:
I see a couple of assumptions here and above, both message and broadcast not being `nil`. I guess that should be the case now but just want to make sure we are also considering the use of this in the future.

Reply:
`message` could be nil, but `broadcast` can't as long as `MakeNewWebSocketEvent` was used to create the event. I can see nil-checking this being good, but I don't think we typically do that for most parameters in the app, so I'd rather not add it here for a case that "shouldn't happen".

I mostly added these functions mostly to give a bit of type safety to this code, and there might be a way to do this by using some generic nonsense (like associating the `broadcastAddMentions` constant with a concrete arguments type) which would avoid needing a new helper function entirely, but that's potentially more scope creep on this PR